rules-reviewed/eap7/eap6/hibernate4.windup.xml

```diff
@@ -604,5 +572,198 @@
+                        auto-register themselves with `org.hibernate.type.descriptor.sql.SqlTypeDescriptorRegistry` and `org.hibernate.type.descriptor.java.JavaTypeDescriptorRegistry`.
+                        Applications using custom `SqlTypeDescriptor`/`JavaTypeDescriptor` implementations which extends the built-in ones and
+                        rely on that behavior should be updated to call `SqlTypeDescriptorRegistry#addDescriptor` or `JavaTypeDescriptorRegistry#addDescriptor` themselves.</message>
+                    <link href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#type-handling"
+                        title="Hibernate 5 Type Handling" />
+                    <link
+                        href="https://docs.jboss.org/hibernate/orm/5.0/javadocs/index.html?org/hibernate/type/descriptor/sql/SqlTypeDescriptorRegistry.html"
+                        title="Hibernate 5 javadoc for SqlTypeDescriptorRegistry#addDescriptor" />
+                    <link href="https://docs.jboss.org/hibernate/orm/5.0/javadocs/index.html?org/hibernate/type/descriptor/java/JavaTypeDescriptorRegistry.html" title="Hibernate 5 javadoc for JavaTypeDescriptorRegistry#addDescriptor" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+            <where param="spatial">
+                <matches
+                    pattern="(mysql.MySQLGeometryTypeDescriptor | h2geodb.GeoDBGeometryTypeDescriptor | postgis.PGGeometryTypeDescriptor | oracle.SDOGeometryTypeDescriptor | sqlserver.SqlServer2008GeometryTypeDescriptor)" />
+            </where>
+            <where param="sqltype">
+                <matches
+                    pattern="(AbstractTypeDescriptor | BigDecimalTypeDescriptor | BigIntegerTypeDescriptor | BlobTypeDescriptor | BooleanTypeDescriptor | ByteArrayTypeDescriptor | ByteTypeDescriptor | CalendarDateTypeDescriptor | CalendarTimeTypeDescriptor | CalendarTypeDescriptor | CharacterArrayTypeDescriptor | CharacterTypeDescriptor | ClassTypeDescriptor | ClobTypeDescriptor | CurrencyTypeDescriptor | DateTypeDescriptor | DoubleTypeDescriptor | DurationJavaDescriptor | EnumJavaTypeDescriptor | FloatTypeDescriptor | InstantJavaDescriptor | IntegerTypeDescriptor | JavaTypeDescriptorRegistry.FallbackJavaTypeDescriptor | JdbcDateTypeDescriptor | JdbcTimestampTypeDescriptor | JdbcTimeTypeDescriptor | LocalDateTimeJavaDescriptor | LocalDateJavaDescriptor | LocaleTypeDescriptor | LocalTimeJavaDescriptor | LongTypeDescriptor | NClobTypeDescriptor | OffsetDateTimeJavaDescriptor | OffsetTimeJavaDescriptor | PrimitiveByteArrayTypeDescriptor | PrimitiveCharacterArrayTypeDescriptor | SerializableTypeDescriptor | ShortTypeDescriptor | StringTypeDescriptor | TimeZoneTypeDescriptor | UrlTypeDescriptor | UUIDTypeDescriptor | ZonedDateTimeJavaDescriptor)" />
```

Comment:
There's only one `AbstractTypeDescriptor` in Hibernate -- in package `org.hibernate.type.descriptor.java`. So this is in wrong `<where>`.